session/session_test.go

```diff
@@ -171,12 +186,28 @@ func (s *testSessionSuiteBase) SetUpSuite(c *C) {
 	s.dom.GetGlobalVarsCache().Disable()
 }
 
+func (s *testSessionSuiteBase) SetUpSuite(c *C) {
+	testleak.BeforeTest()
+
+	// TODO: if we run `-with-tikv -check.p true` this will cause deadlock, due to
+	// https://github.com/pingcap/check/blob/8a5a85928f125d818621be7f0ee69d7207f53622/check.go#L646
```

Comment:
Does this mean that if we handle the code of "check.go#L646", it couldn't cause deadlock?
If so, you do it in [PR](https://github.com/pingcap/check/pull/17). I think we can remove this comment.

Reply:
yes, that pr is used for handling this problem. But currently, it will cause some test suite fail. I'm working on fixing these tests.